MvvmCross.Droid.Support.V7.RecyclerView/MvxRecyclerView.cs

```diff
@@ -44,6 +46,32 @@ public MvxRecyclerView(Context context, IAttributeSet attrs, int defStyle, IMvxR
                 ItemTemplateId = itemTemplateId;
         }
 
+        public override void OnStartTemporaryDetach()
+        {
+            _temporarilyDetached = true;
+            base.OnStartTemporaryDetach();
+        }
+
+        public override void OnFinishTemporaryDetach()
+        {
+            _temporarilyDetached = false;
+            base.OnFinishTemporaryDetach();
+        }
+
+        protected override void OnDetachedFromWindow()
+        {
+            base.OnDetachedFromWindow();
+
+            if (!_temporarilyDetached)
+            {
+                // Clear out all bindings from the ViewHolders created by the current adapter.
+                // This is for https://github.com/MvvmCross/MvvmCross/issues/1379
+                // According to the documentation Adapters can be owned by multiple
+                // RecyclerViews so this might not be entirely accurate.
+                this.Adapter?.ClearAllBindings();
```

Comment:
The framework itself will create a new instance, so in those cases it shouldn't matter. However, the user can create their own instance and share it. How often that will be the case I don't know.

There is also https://developer.android.com/reference/android/support/v7/widget/RecyclerView.OnChildAttachStateChangeListener.html which might be useful to use instead, so the ViewHolder itself does not call the Adapter, but the other way around.